infra/terraform/modules/data_access/github_webhooks_handler/tests/conftest.py

```diff
@@ -0,0 +1,140 @@
+from hashlib import sha1
+from moto import mock_sns
+import boto3
+import hmac
+import json
+import os
+
+import pytest
+
+
+#################
+#  Tests state  #
+#################
+
+
+@pytest.fixture
+def state():
+    return {
+        "event": {
+            "headers": {
+                "X-Hub-Signature": "signature",
+                "X-GitHub-Event": "team",
+            },
+            "body": "{\"a JSON string\": true}",
+        }
+    }
+
+
+###########
+#  Event  #
+###########
+
+
+@pytest.fixture
+def event(state):
+    return state["event"]
+
+
+@pytest.fixture
+def event_type(event):
+    return event["headers"]["X-GitHub-Event"]
+
+
+###############
+#  Signature  #
+###############
+
+
+@pytest.fixture
+def given_invalid_signature(state):
+    state["event"]["headers"]["X-Hub-Signature"] = "sha3=invalid"
+
+
+@pytest.fixture
+def given_valid_signature(state, secret):
+    message = "hello this is dog"
+    signature = sha1_signature(message, secret)
+
+    state["event"]["body"] = message
+    state["event"]["headers"]["X-Hub-Signature"] = signature
+
+
+def sha1_signature(message, secret):
+    mac = hmac.new(
+        secret.encode("utf-8"),
+        msg=message.encode("utf-8"),
+        digestmod=sha1
+    )
+    return "sha1={}".format(mac.hexdigest())
+
+
+############
+#  Config  #
+############
+
+
+@pytest.fixture
+def given_the_env_is_set(sns_arn, secret):
+    os.environ["SNS_ARN"] = sns_arn
+    os.environ["STAGE"] = "test"
+    os.environ["GH_HOOK_SECRET"] = secret
```

Comment:
This will set environment variables when the tests are run. I'm sure it's not a problem, but tests should really be idempotent.

Later tests might depend on the environment being unchanged (they shouldn't of course), so it's best practice not to change the state of the system.